acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/pages/PageHelper.java

```diff
@@ -71,48 +67,53 @@ private WebElement findElement(final Function<By, List<WebElement>> findElements
     public void waitUntilTrue(final Predicate predicate) {
 
         pollWithTimeout().until(webDriver -> {
-                try {
-                    return predicate.executeWithPredicate();
-                } catch (final Exception e) {
-                    return false;
-                }
-            });
+            try {
+                return predicate.executeWithPredicate();
+            } catch (final Exception e) {
+                return false;
+            }
+        });
     }
 
     public void executeWhenStable(final Task task) {
 
         pollWithTimeout().until(webDriver -> {
-                try {
-                    task.execute();
-                    return true;
-                } catch (final Exception e) {
-                    return false;
-                }
-            });
+            try {
+                task.execute();
+                return true;
+            } catch (final Exception e) {
+                return false;
+            }
+        });
     }
 
     public void waitUntilVisible(WebElement webElement) {
-        try{
+        try {
             pollWithTimeout().until((WebDriver innerDriver) -> webElement.isDisplayed());
         }
         catch (TimeoutException e){
             throw new RuntimeException("Timeout while waiting for element to become visible: '" + webElement + "'", e);
         }
     }
 
+    /**
+     * This takes care of element being removed/refreshed from DOM
+     */
+    public <T> T waitForElementUntil(ExpectedCondition<T> condition) {
+        return pollWithTimeout()
+            .until(ExpectedConditions.refreshed(condition));
```

Comment:
The `ExpectedCondition` is a really nice find! Had I known about it sooner, perhaps I'd have avoided adding the two functional interfaces at the end of this class and the methods that use them.